From 2571b979d5fcadcdd458df2ecf1f525a5955a80a Mon Sep 17 00:00:00 2001
From: Zhang Ming <154842638(a)qq.com>
Date: Wed, 17 Mar 2021 21:46:41 +0800
Subject: [PATCH] arm64/mpam: fix a memleak in add_schema
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 2571b979d5fcadcdd458df2ecf1f525a5955a80a


openEuler inclusion
category: bugfix
bugzilla: 48265
CVE: NA
Reference: https://gitee.com/openeuler/kernel/issues/I3BPPX

---------------------------------------------------

The default branch in switch will not run at present,
but there may be related extensions in the future,
which may lead to memory leakage.

Signed-off-by: Zhang Ming <154842638(a)qq.com>
Reported-by: Wang ShaoBo <bobo.shaobowang(a)huawei.com>
Suggested-by: Jian Cheng <cj.chengjian(a)huawei.com>
Reviewed-by: Xie XiuQi <xiexiuqi@huawei.com>
[Zheng Zengkai: adjust commit message]
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/arm64/kernel/mpam/mpam_ctrlmon.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm64/kernel/mpam/mpam_ctrlmon.c b/arch/arm64/kernel/mpam/mpam_ctrlmon.c
index 4acf9234c3a5..b1d32d432556 100644
--- a/arch/arm64/kernel/mpam/mpam_ctrlmon.c
+++ b/arch/arm64/kernel/mpam/mpam_ctrlmon.c
@@ -74,6 +74,7 @@ static int add_schema(enum resctrl_conf_type t, struct resctrl_resource *r)
 		suffix = "";
 		break;
 	default:
+		kfree(s);
 		return -EINVAL;
 	}
 
-- 
2.26.2

